web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -259,6 +259,12 @@ public String getFullRecord(String pidString, Model model, HttpServletRequest re
             // Get the file to download
             String dataFileUrl = accessCopiesService.getDownloadUrl(briefObject, principals);
             model.addAttribute("dataFileUrl", dataFileUrl);
+
+            // Get thumbnail object
+            ContentObjectRecord thumbnailObject = accessCopiesService.getThumbnailObject(briefObject, principals);
```

Comment:
Rather than make this determination in the FullRecordController, I think we should move this over to the `/services/api/thumb/${pid}` endpoints. The reason being that we will probably need to embed thumbnails into finding aids and other sites, which will be making requests directly against the services api, and for archival folders it will be pointing at the work for the top level thumbnail I believe. If you move the logic over to there, then I don't think any changes will be needed in the public UI directly.

Reply:
Would all these calls be to the file object? Moving the check over seems to break file level thumbnails and shows the wrong thumbnail for works without primary objects since the thumbnail brief object is whatever the original is
```
<c:set var="thumbnailObject" value="${thumbnailObject}" scope="request" />
<c:import url="common/thumbnail.jsp">
      <c:param name="target" value="file" />
      <c:param name="size" value="large" />
</c:import>
```

Reply:
I think you'll need to add a check up front in `getThumbnailObject` to see if the object being requested is a WorkObject. If it is, then you would need to do the logic to check for a primary object and fall back to querying for children with thumbnails. For any other type (FileObjects, Collections with thumbnails, etc), you would only want to check the object itself.

This is a functionality change for the thumbnail endpoint, which previously didn't work for WorkObjects directly. So I guess we will need to make changes in the public UI to get it to request the Works thumbnail rather than its primary objects. We would want to switch to this behavior for search results too. I'm not quite sure what the impact of this change will be on how the placeholders get displayed. It might still work?